src/aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -198,6 +198,10 @@ def get_builder_from_protocol(
             parameter_overrides = overrides.get('pw', {}).get('parameters', {})
             parameters = recursive_merge(parameters, parameter_overrides)
 
+            # if tot_magnetization in overrides , remove starting_magnetization from parameters
+            if overrides['pw']['parameters']['SYSTEM'].get('tot_magnetization') is not None:
+                parameters['SYSTEM'].pop('starting_magnetization', None)
```

Comment:
You need to guard against `SYSTEM` not yet being defined.
```suggestion
            # if tot_magnetization in overrides , remove starting_magnetization from parameters
            if parameters.get('SYSTEM', {}).get('tot_magnetization') is not None:
                parameters.setdefault('SYSTEM', {}).pop('starting_magnetization', None)
```

Reply:
@sphuber Thank you so much for the help! , I will also continue checking the PdosWorkChain and BandsWorkChain, since somehow it produces a problem of not having a fermi_energy